includes/polling/os/aruba-instant.inc.php

```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * aruba-instant.inc.php
+ *
+ * LibreNMS os polling module for Aruba Instant
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @package    LibreNMS
+ * @link       http://librenms.org
+ * @copyright  2019 Timothy Willey
+ * @author     Timothy Willey <developer@timothywilley.net>
+ */
+use LibreNMS\RRD\RrdDefinition;
+
+// ArubaOS (MODEL: 225), Version 8.4.0.0-8.4.0.0
+// ArubaOS (MODEL: 105), Version 6.4.4.8-4.2.4.12
+$badchars                    = array( '(', ')', ',',);
+list(,,$hardware,,$version,) = str_replace($badchars, '', explode(' ', $device['sysDescr']));
+
+$client_count = 0;
+// version 8.4.0.0 introduced client counts per radio as well as per ssid.
+// prior versions require iteration over the list of clientss to get a count.
+if (intval(explode('.', $version)[0]) >= 8 && intval(explode('.', $version)[1]) >= 4) {
+    // ArubaOS >= 8.4.0.0
+    // sum the values values of aiRadioClientNum or aiSSIDClientNum to count clients
+    $client_data = snmpwalk_group($device, 'aiSSIDClientNum', 'AI-AP-MIB');
+    d_echo('Debug Instant aiSSIDClientNum: '.PHP_EOL);
+    d_echo(var_export($client_data, 1));
+    foreach ($client_data as $key => $value) {
+        $client_count += intval($value['aiSSIDClientNum']);
+    }
+} else {
+    // ArubaOS =< 8.4.0.0
+    // count number of aiClientMACAddress or aiClientWlanMACAddress
+    $client_data = snmpwalk_group($device, 'aiClientMACAddress', 'AI-AP-MIB');
+    d_echo('Debug Instant aiClientMACAddress: '.PHP_EOL);
+    d_echo(var_export($client_data, 1));
+    foreach ($client_data as $key => $value) {
+        $client_count += 1;
+    }
+}
+
+$ap_count = 0;
+// Count the number of AP MAC addresses (could use AP IPs, Serial Number, etc too.)
+$ap_data = snmpwalk_group($device, 'aiAPMACAddress', 'AI-AP-MIB');
```

Comment:
This should be done within the wireless code.
Unifi does it for example.
If you can't do it, the wireless module supports custom polling code.

Reply:
I'm trying to get a total clients and total aps count for the overview graphs. there's no single oid that has the total number of clients or aps in the cluster, so i have to calculate the totals. is that possible using the wireless sensors code?

Reply:
Ok, I'm digging into this, and see how it can be done in the wireless code. Doesn't look too difficult.

Reply:
You need to implement `\LibreNMS\Interfaces\Polling\Sensors\WirelessClientsPolling` on your ArubaInstant model.

Then you can basically just put all this code inside the `pollWirelessClients` function